packages/babel-plugin-transform-async-to-promises/src/index.js

```diff
@@ -0,0 +1,113 @@
+/* Babel 7 transform */
+
+const babylon = require("babylon");
+const generate = require("@babel/generator").default;
+
+function printNode(ast) {
+  return generate(ast, {}).code;
+}
+
+function parse(code) {
+  return babylon.parse(code, {
+    allowImportExportEverywhere: true,
+    allowReturnOutsideFunction: true,
+    allowSuperOutsideMethod: true,
+  }).program;
+}
+
+const transform = require("nodent-transform").transform;
+
+function transformAsyncToPromises(api, options) {
+  let requiresTranspilation;
+  const runtime = options.runtime ? options.runtime : null;
+  const opts = {
+    // Code generation options
+    es6target: false,
+    babelTree: true,
+    engine: false,
+    generators: false,
+    promises: true,
+    lazyThenables: false,
+    wrapAwait: true,
+    noRuntime: !runtime,
+    $runtime: runtime,
+    generatedSymbolPrefix: "$",
+    $return: "$return",
+    $error: "$error",
+    $arguments: "$args",
+    $Promise: "Promise",
+    $asyncspawn: "$asyncspawn",
+    $asyncbind: "$asyncbind",
+    $makeThenable: "$makeThenable",
+  };
+
+  if (options.codeGenerationOptions) {
+    const keys = Object.keys(options.codeGenerationOptions);
+    for (let i = 0; i < keys.length; i++) {
+      if (keys[i] in opts) {
```

Comment:
This check should probably be `Object.prototype.hasOwnProperty.call`, in case the key is “toString” or something.